### ISSUE TITLE
Height of charts keeps increasing endlessly when using CSS grid or Flexbox

### ISSUE THREAD

**Issue by Snedden** [plouc/nivo#867]
**Describe/explain the bug**
The pie chart keeps on increasing its height inside a sematic flex grid.

**To Reproduce**
Have a pie chart inside a sematic UI flex grid 
Here is an example in sand box :https://codesandbox.io/s/semantic-ui-example-woriq?fontsize=14&hidenavigation=1&theme=dark

**Expected behavior**
The svg rect  height should be maxed at height of the container

**Desktop (please complete the following information):**
 - Browser [chrome]
 - Version [ 79.0.3945.130 (Official Build) (64-bit)]

**Smartphone (please complete the following information):**
NA

**Comment by OllyHodgson** [comment]
As a workaround, I had to set a `height` in the container's CSS. `max-height` and `flex-basis` did not appear to have the desired effect.

This issue seems to affect the responsive Pie, Bar and Line components. I've not tried the others, yet.

**Comment by thedaviddias** [comment]
Same issue, but I'm using theme-ui...

**Comment by vdh** [comment]
It's still a current issue

**Comment by bengraves-five** [comment]
bump

**Comment by astroanu** [comment]
any idea why this is happening ?

**Comment by siavhnz** [comment]
same problem no solution :(

**Comment by jrussellfreelance** [comment]
Yeah I was excited to use this software but also encountered this.

**Comment by LoicUV** [comment]
Probably the same issue as https://github.com/plouc/nivo/issues/109. Maybe try the [workaround](https://github.com/plouc/nivo/issues/109#issuecomment-464367293).

**Comment by hayriyigit** [comment]
bump

**Comment by vasilica33** [comment]
Same issue with bump

**Comment by greengiraffe** [comment]
The bug is still present at v0.73.1

See https://codesandbox.io/s/semantic-ui-example-forked-ps9x9

**Comment by greengiraffe** [comment]
This workaround was already suggested above: https://github.com/plouc/nivo/issues/867#issuecomment-638993290
It's not useful if the chart should scale dynamically with its parent, as is often desired when using flexbox.

**Comment by marcospassos** [comment]
Same here.

**Comment by OllyHodgson** [comment]
As far as I can tell, this is still an issue. See https://github.com/plouc/nivo/issues/867#issuecomment-638993290 for a workaround.

**Comment by marcospassos** [comment]
I can confirm that.

**Comment by malte94** [comment]
Still an issue.

**Comment by maxizhukov** [comment]
Still same issue and when I set:

```
<div style={{position: "relative"}}>
        <div style={{ position: "absolute", width: "100%", height: "400px" }}>
             <ResponsiveLine />
        </div>
</div>
```

There are no chart inside, just blank

**Comment by cutecycle** [comment]
Affects me as well in `ResponsiveNetwork`

**Comment by joseph-mccombs** [comment]
bump

**Comment by azeemakhter-arbisoft** [comment]
bump

**Comment by hyukkyukang** [comment]
bump

**Comment by anvaynk** [comment]
bump bump bump, are you folks gonna fix this or what ?

**Comment by joesaunderson** [comment]
Bump on this

**Comment by bradlocking** [comment]
Bump on this - making it incredibly difficult to use modern layouts alongside any nivo chart components

**Comment by JayYoungman** [comment]
same issue.  Poor support this hasn't been fixed yet.

**Comment by JayYoungman** [comment]
If anyone else is still struggling with this, this is my solution, far from ideal but it solves the issue. 

`<Pane
          background="white"
          display="flex"
          elevation={1}
          flexDirection="column"
          height={channelsData.length * 54}
          justifyContent="top"
          padding={majorScale(2)}
          width="100%"
        >
          <Text>Channels &amp; Listings</Text>
          <Pane height="100%">
            <BarChart
              channelsData={channelsData}
              numOfProducts={numOfProducts}
            />
          </Pane>
        </Pane>`

Notice the height calculated based on the length of my data, which in my case could be 100s

**Comment by marcospassos** [comment]
bump

**Comment by malte94** [comment]
bump

**Comment by AndreiManya** [comment]
bump

**Comment by ThibaudMoretto** [comment]
bump

**Comment by jackLIT** [comment]
bump

**Comment by skillrepository** [comment]
bump

**Comment by Kritik-Kapoor** [comment]
bump

**Comment by LinusHe** [comment]
bump

**Comment by ozansozuozgit** [comment]
bump

**Comment by joelbyrd** [comment]
bump

**Comment by narekmargaryan** [comment]
I've tried a couple of older suggestions to no avail. What I've found to work is wrapping the chart in a div container with 100% width and 1/1 aspect ratio:

```typescript
<div
  style = {{
      width: "100%",
      aspectRatio: "1/1",
  }}
>
    <ResponsivePie/>
</div>
```

[CodeSandBox example](https://codesandbox.io/embed/quirky-colden-6tz463?fontsize=14&hidenavigation=1&theme=dark)

Cheers.

**Comment by JaguarRocka** [comment]
I can provisory fix it with use `height={300}` directly as value in my ReponsivePie : 
 <ResponsivePie
    data={countVersions}
     height={300}
     margin={{
     ...
  />

**Comment by doominkim** [comment]
Hasn't it been solved yet?

**Comment by dudiharush** [comment]
@doominkim use the regular component, not the responsive one, and add responsiveness with react-virtualized or react-window. It worked for me :)

**Comment by endymion** [comment]
Nivo would be a lot better if it had some documentation on this.

**Comment by sam-keen** [comment]
@plouc please can you document this, it's not clear how to achieve what you've managed on the Nivo site. Many thanks

**Comment by Todaricli** [comment]
bump

**Comment by JosephReece** [comment]
bump

**Comment by plouc** [comment]
I've pinned this issue as it's clearly a bug, I don't have  a fix atm, but properly supporting flexbox/grid is indeed a requirement.
